argoverse/evaluation/stereo/utils.py

```diff
@@ -0,0 +1,160 @@
+
+    for abs_error_thresh, rel_error_thresh in zip(abs_error_thresholds, rel_error_thresholds):
+        bad_pixels = (abs_err > abs_error_thresh) & (rel_err > rel_error_thresh)
+
+        errors[f"num_errors_bg:{abs_error_thresh}"] = np.sum(bg_mask & bad_pixels)
+        errors[f"num_errors_fg:{abs_error_thresh}"] = np.sum(fg_mask & bad_pixels)
+        errors[f"num_errors_bg_est:{abs_error_thresh}"] = np.sum(bg_mask & pred_mask & bad_pixels)
+        errors[f"num_errors_fg_est:{abs_error_thresh}"] = np.sum(fg_mask & pred_mask & bad_pixels)
+
+    if save_disparity_error_image:
+        err = np.minimum(abs_err / abs_error_thresholds[0], rel_err / rel_error_thresholds[0])
+        disparity_error_image = np.zeros((*pred_disparity.shape, 3), dtype=np.uint8)
+
+        for threshold, color in LOG_COLORMAP:
+            disparity_error_image[np.logical_and(err >= threshold[0], err < threshold[1])] = color
+
+        disparity_error_image[gt_disparity == 0] *= 0
+
+        disparity_error_image = np.uint8(
+            cv2.dilate(disparity_error_image, kernel=np.ones((2, 2), np.uint8), iterations=3)
+        )
+
+        # Plot the custom log-colormap and blend it with the disparity error image.
+        fig = plt.figure(figsize=(24, 2))
+        ax = fig.add_subplot(111)
+        x = np.linspace(1, 11, 10)
+        y = np.linspace(0, 0, 10)
+        scalars = np.array(LOG_COLORMAP, dtype=object)[:, 0] * abs_error_thresholds[0]
+        scalars = [f"[{scalar[0]:.2f}, {scalar[1]:.2f}]" for scalar in scalars]
+        colors = np.array(LOG_COLORMAP, dtype=object)[:, 1]
+        colors = [color / 255.0 for color in colors]
+
+        # Plot the custom log-colormap as colored circles with the disparity error ranges as labels.
+        plt.scatter(x, y, c=colors, s=3000)
+        plt.xlabel("Disparity range errors in pixels", fontsize=24)
+        plt.xticks(x, scalars, fontsize=20)
+        plt.yticks([])
+
+        # Remove the plot frame to make the log-colormap more clear.
+        ax.spines["top"].set_visible(False)
+        ax.spines["bottom"].set_visible(False)
+        ax.spines["right"].set_visible(False)
+        ax.spines["left"].set_visible(False)
+        plt.tight_layout()
+
+        # Recover the log-colormap drawing as a np.array image.
+        fig.canvas.draw()
+        colormap_image = np.frombuffer(fig.canvas.tostring_rgb(), dtype=np.uint8)
+        colormap_image = colormap_image.reshape(fig.canvas.get_width_height()[::-1] + (3,))
+        plt.close()
+
+        # Blend the log-colormap image to the top of the disparity error image.
+        c_img = colormap_image[:, :, ::-1]
+        d_img = disparity_error_image[:, :, ::-1]
+        x_offset = 30
+        y_offset = 30
+        d_img[y_offset : y_offset + c_img.shape[0], x_offset : x_offset + c_img.shape[1]] = c_img
+
+        # Save the blended disparity error image to a PNG file.
+        log_id = str(gt_fpath).split("/")[-3]
```

Comment:
perhaps simpler as `Path(gt_fpath).parts[-3]`

Reply:
This is cool! much cleaner.